### PR TITLE
Runtime core swap

### PR DESCRIPTION
**Pull request by junmendoza**
@ke-yu @aparajit-pratap 

1. Moving FFI runtime properties to runtimeCore
2. Move Runningblock to runtimeCore

### CODE REVIEW

File: src/Engine/ProtoCore/Core.cs [hunk 1]
```diff
@@ -802,20 +759,6 @@ public bool CancellationPending
         public int FunctionCallDepth { get; set; }
         public TextWriter ExecutionLog { get; set; }
 
-        protected void OnDispose()
-        {
-            if (Dispose != null)
-            {
-                Dispose(this);
```
Comment (ke-yu) [inline src/Engine/ProtoCore/Core.cs]: Dispose() is a event handler, make sure if it is safe to delete `OnDispose()`

File: test/Engine/ProtoTestFx/DebugService.cs [hunk 2]
```diff
@@ -136,6 +139,7 @@ public bool DebugScript(string dsPath)
             System.IO.StringWriter stringStream = new StringWriter();
             executionLog = new StringBuilder();
             ProtoCore.Core core = null;
+            ProtoCore.RuntimeCore runtimeCore = null;
```
Comment (ke-yu) [inline test/Engine/ProtoTestFx/DebugService.cs]: Move runtimeCore to `try` block.

File: test/Engine/ProtoTestFx/DebugService.cs [hunk 3]
```diff
@@ -70,8 +71,10 @@ public bool RunScript(string dsPath)
                 core.Options.RootModulePathName = ProtoCore.Utils.FileUtils.GetFullPathName(dsPath);
                 core.Compilers.Add(ProtoCore.Language.kAssociative, new ProtoAssociative.Compiler(core));
                 core.Compilers.Add(ProtoCore.Language.kImperative, new ProtoImperative.Compiler(core));
-                core.Configurations.Add(Autodesk.DesignScript.Interfaces.ConfigurationKeys.GeometryFactory, GeometryFactoryName);
-                core.Configurations.Add(Autodesk.DesignScript.Interfaces.ConfigurationKeys.PersistentManager, PersistenceManagerName);
+
+                runtimeCore = core.__TempCoreHostForRefactoring;
```
Comment (ke-yu) [inline test/Engine/ProtoTestFx/DebugService.cs]: core contains a runtime core or we finally will fully split them?

**Comment by ke-yu** [comment]
@junmendoza , I'm not quite sure about the design decision - that `__TempCoreHostForRefactorign` will finally removed, right? And will `Core` and `RuntimeCore` be fully split? If a core still contains a `RuntimeCore`, IMHO, it is just a wrapper to wrap a chunk of code that previously was in core, but doesn't have much benefit. Is my understanding wrong?

File: test/Engine/ProtoTestFx/DebugService.cs (hunk 3 above)
Comment (junmendoza) [inline test/Engine/ProtoTestFx/DebugService.cs]: @ke-yu yes, Core and RuntimeCore will be full split and completely independent of each other.
__TempCoreHostForRefactorign will be completely removed 

So essentially visualize this:
    Core core = Compile(sourcecode)
    RuntimeCore runtimeCore = new RuntimeCore (core.RuntimeData)
    Mirror = runtimeCore.Execute()

**Comment by junmendoza** [comment]
@ke-yu yes, Core and RuntimeCore will be full split and completely independent of each other.
__TempCoreHostForRefactorign will be completely removed

So essentially visualize this:
Core core = Compile(sourcecode)
RuntimeCore runtimeCore = new RuntimeCore (core.RuntimeData)
Mirror = runtimeCore.Execute()

**Comment by ke-yu** [comment]
OK. Please make sure it is safe for that `Dispose()` event handler -- some client may relies on it to do some cleanup work, otherwise it is LGTM.

File: src/Engine/ProtoCore/Core.cs (hunk 1 above)
Comment (junmendoza) [inline src/Engine/ProtoCore/Core.cs]: Sure, OnDispose is just moved to rutnimecore. it is still called

**Pull request merged**